src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/TradeQuotes.tsx

```diff
@@ -93,14 +156,71 @@ export const TradeQuotes: React.FC<TradeQuotesProps> = memo(({ isLoading, onBack
     isTradeQuoteRequestAborted,
     bestQuoteData?.quote,
     bestQuoteData?.inputOutputRatio,
-    tradeQuoteDisplayCache,
+    validTradeQuotesDisplayCache,
     activeQuoteMeta,
     isTradeQuoteApiQueryPending,
     isLoading,
     isSwapperQuoteAvailable,
     onBack,
   ])
 
+  const invalidQuotes = useMemo(() => {
+    if (isTradeQuoteRequestAborted) {
+      return []
+    }
+
+    const bestTotalReceiveAmountCryptoPrecision = bestQuoteData?.quote
+      ? getBuyAmountAfterFeesCryptoPrecision({
+          quote: bestQuoteData.quote,
+        })
+      : undefined
+
+    return invalidTradeQuotesDisplayCache.map((quoteData, i) => {
+      const { swapperName, id, errors, isStale } = quoteData
+
+      const isActive = activeQuoteMeta !== undefined && activeQuoteMeta.identifier === id
+
+      return (
+        <TradeQuote
+          isActive={isActive}
+          isLoading={Boolean(isTradeQuoteApiQueryPending[quoteData.swapperName]) && isStale}
+          isRefetching={
+            isLoading ||
+            isTradeQuoteApiQueryPending[quoteData.swapperName] ||
+            !isSwapperQuoteAvailable[swapperName]
```

Comment:
suggestion: This logic seems to be duplicated for the valid ones too, can we make a helper function so they cannot diverge?

Reply:
Changed to a `useCallback` (for simplicity, until it's reused somewhere else) in [2c0e7de](https://github.com/shapeshift/web/pull/8087/commits/2c0e7ded59e68034b730c8481c951c1f9dbf0bfa)